community/index/src/main/java/org/neo4j/index/gbptree/GBPTree.java

```diff
@@ -0,0 +1,481 @@
+                }
+                // This index was created with another page size, re-open with that actual page size
+                if ( pageSize != pageCache.pageSize() )
+                {
+                    if ( pageSize > pageCache.pageSize() )
+                    {
+                        throw new IllegalStateException( "Index was created with page size:" + pageSize
+                                + ", but page cache used to open it this time has a smaller page size:"
+                                + pageCache.pageSize() + " so cannot be opened" );
+                    }
+                    pagedFile.close();
+                    pagedFile = pageCache.map( indexFile, pageSize );
+                }
+                return pagedFile;
+            }
+            catch ( Throwable t )
+            {
+                try
+                {
+                    pagedFile.close();
+                }
+                catch ( IOException e )
+                {
+                    t.addSuppressed( e );
+                }
+                throw t;
+            }
+        }
+        catch ( NoSuchFileException e )
+        {
+            pageSize = pageSizeForCreation == 0 ? pageCache.pageSize() : pageSizeForCreation;
+            if ( pageSize > pageCache.pageSize() )
+            {
+                throw new IllegalStateException( "Index was about to be created with page size:" + pageSize +
+                        ", but page cache used to create it has a smaller page size:" +
+                        pageCache.pageSize() + " so cannot be created" );
+            }
+
+            // We need to create this index
+            PagedFile pagedFile = pageCache.map( indexFile, pageSize, StandardOpenOption.CREATE );
+
+            // Write header
+            try ( PageCursor metaCursor = openMetaPageCursor( pagedFile ) )
+            {
+                metaCursor.putInt( pageSize );
+                metaCursor.putLong( rootId );
+                metaCursor.putLong( lastId );
+                metaCursor.putLong( layout.identifier() );
+                metaCursor.putInt( layout.majorVersion() );
+                metaCursor.putInt( layout.minorVersion() );
+                layout.writeMetaData( metaCursor );
+            }
+            pagedFile.flushAndForce();
+            created = true;
+            return pagedFile;
+        }
+    }
+
+    private PageCursor openMetaPageCursor( PagedFile pagedFile ) throws IOException
+    {
+        PageCursor metaCursor = pagedFile.io( META_PAGE_ID, PagedFile.PF_SHARED_WRITE_LOCK );
+        if ( !metaCursor.next() )
+        {
+            throw new IllegalStateException( "Couldn't go to meta data page " + META_PAGE_ID );
+        }
+        return metaCursor;
+    }
+
+    @Override
+    public RawCursor<Hit<KEY,VALUE>,IOException> seek( KEY fromInclusive, KEY toExclusive ) throws IOException
+    {
+        long localRootId = rootId;
```

Comment:
i do not think we need localRootId variable in a for it's used now, it's only brings additional confusion why it's need to be reassigned.